runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java

```diff
@@ -99,6 +100,29 @@ public SimpleMonitoringInfoBuilder setInt64Value(long value) {
     return this;
   }
 
+  /**
+   * Sets the IntDistributionData of the DistributionData in the MonitoringInfo, and the appropriate
+   * type URN.
+   */
+  public SimpleMonitoringInfoBuilder setInt64DistributionValue(DistributionData data) {
+    this.builder
+        .getMetricBuilder()
+        .getDistributionDataBuilder()
+        .getIntDistributionDataBuilder()
+        .setCount(data.count())
+        .setSum(data.sum())
+        .setMin(data.min())
+        .setMax(data.max());
+    this.setInt64DistributionTypeUrn();
+    return this;
+  }
+
+  /** Sets the the appropriate type URN for int64 distribution tuples. */
+  public SimpleMonitoringInfoBuilder setInt64DistributionTypeUrn() {
```

Comment:
If the monitoring info type is set by the `set..Value` methods, I wonder if the `set..TypeUrn` methods should be private?

Reply:
We call it in some test cases.
Its nice to test those scenarios because its not obvious how maintainers will make changes. The test specifically makes sure that if the value is not set (even if the type is) then we won't validate the metrics. Its important to make sure the validators will still work properly in that scenario.